hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -587,12 +617,25 @@ private void initialize() {
 
   @Override
   public void startup() {
-    // mark we are running now
-    this.sourceRunning = true;
-    initThread = new Thread(this::initialize);
-    Threads.setDaemonThreadRunning(initThread,
-      Thread.currentThread().getName() + ".replicationSource," + this.queueId,
-      this::uncaughtException);
+    //Flag that signalizes uncaught error happening while starting up the source
+    // and a retry should be attempted
+    AtomicBoolean retryStartup = new AtomicBoolean(false);
+    retryStartup.set(true);
+    do {
+      if(retryStartup.get()) {
+        retryStartup.set(false);
+        // mark we are running now
+        this.sourceRunning = true;
+        initThread = new Thread(this::initialize);
+        Threads.setDaemonThreadRunning(initThread,
+          Thread.currentThread().getName() + ".replicationSource," + this.queueId,
+          (t,e) -> {
+          sourceRunning = false;
+          uncaughtException(t, e, null, null);
+          retryStartup.set(true);
+        });
+      }
+    } while (!this.sourceRunning);
```

Comment:
I had a second thought on this here, we can't simply re-use this boolean, because in case of failure, we risk reach this point before the exception handler has updated it to false. I'm bringing back the original _startupOngoing_ in the next commit,